### DISCUSSION TITLE
Python3.11 and later frame access

### DISCUSSION THREAD

**Discussion by dimaqq** [PyO3/pyo3#4759] [Questions]
Back in 3.9, I used `frame->f_stacktop` but that got removed in 3.10 and that's reflected in the rust bindings.

Then (for some other purpose) I used `frame->f_valuestack` but got got removed in 3.11 together with the frame struct reorganisation (the struct is effectively private), and that too is reflected in the rust bindings.

3.11 did bring in an alternative API: PyFrame_GetLocals, etc. I don't see this in the rust bindings, but maybe I'm not looking in the right place?

What's the PyO3 recommended way to poke into the Python stack?

**Comment by davidhewitt** [comment]
Created #4762 - help welcome to expose these APIs.
